### DISCUSSION TITLE
Output Not Appearing in Debug Console or Terminal

### DISCUSSION THREAD

**Discussion by karthick-govindaraj** [WebFreak001/code-debug#445] [General]
I have multiple print statements in my C code, but the output does not appear in the debug console or terminal when breakpoints are hit. I've attached my launch.json configuration below. Can someone suggest how to ensure that print statements are logged correctly when debugging?

```
{
  "version": "0.2.0",
  "configurations": [
    {
      "name": "Debug C Program",
      "type": "gdb",
      "request": "launch",
      "target": "${workspaceFolder}/hello",
      "cwd": "${workspaceFolder}",
      "preLaunchTask": "build hello.c",
      "stopAtEntry": true
    }
  ]
}
```

![image](https://github.com/user-attachments/assets/bb7f4e02-914b-4e87-9b91-37888288f309)

**Comment by HenryRiley0** [comment]
```json
{
  "version": "0.2.0",
  "configurations": [
    {
      "name": "Debug C Program",
      "type": "gdb",
      "request": "launch",
      "target": "${workspaceFolder}/hello",
      "cwd": "${workspaceFolder}",
      "preLaunchTask": "build hello.c",
      "stopAtEntry": true,
      "terminal": ""
    }
  ]
}
```

Try adding `"terminal": ""`, which will invoke the system's terminal, and the program's input and output will be in the new terminal. I will fix the issue where the default terminal does not display output and cannot accept input.

**Reply by GitMensch** [reply]
> I will fix the issue where the default terminal does not display output and cannot accept input.

That's cool. Actually I think there should be another setting "internal" which creates a vscode terminal session and has all program screeen i/o there (should work for all launch requests, if we can route the io maybe even including ssh), likely via `vscode.window.createTerminal()`. That may even be a reasonable default as that's unspecific to the OS...

**Reply by karthick-govindaraj** [reply]
Thanks, @HenryRiley0 .

After adding "terminal": "" in launch.json, the system terminal opens at each breakpoint. Is there any other option to display the values in the VS Code terminal instead? If this is an issue, could you please update it ASAP? This feature would be helpful for many developers.
